example/provider-extensions/shoot/gcp.yaml

```diff
@@ -0,0 +1,59 @@
+kind: Shoot
```

Comment:
I don't know whether we really want to maintain this in the gardener/gardener repo. Can't we rather leave this for the user and point to the respective extension repo (which should have proper example files)?

Same for the other examples (secret, binding, ctrlreg, ...)

Reply:
I don't have a strong opinion here. My initial idea was creating something similar to the [provider-local shoot](https://github.com/gardener/gardener/blob/master/example/provider-local/shoot.yaml).
However, it might be too complicated setting up the secrets etc. in a way that they work with these examples 🤔

Reply:
Yes, however, we don't want to maintain provider specifics in g/g. You can put those examples to the respective provider repos so that they can be picked from there :)

Reply:
I deleted the demo shoot configurations.